app_vdi.py

```diff
@@ -0,0 +1,212 @@
+import base64
+import io
+import numpy as np
+import pandas as pd
+import dash
+from dash.dependencies import Output, Input, State
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objs as go
+
+
+# Initializes dash app
+app = dash.Dash(__name__)
+
+app.title = 'vdi visualisation'
+
+VDI_DATA = {}
+
+
+def html_param_input(p_name, p_value=0., p_unit='', **kwargs):
+    """Create a nice html component for parameter inputs.
+
+    :param p_name: (str) the label displayed to the left of the parameter input
+    :param p_value: (number) the initial value of the parameter
+    :param p_unit: (str) the unit displayed to the right of the parameter input
+    :param kwargs: optional arguments for dcc.Input
+    :return: a html.Div instance
+    """
+
+    p_label_name = p_name
+    p_name = p_name.lower()
+
+    return html.Div(
+        id='{}-div'.format(p_name),
+        className='app__input',
+        children=[
+            html.Div(
+                id='{}-label'.format(p_name),
+                className='app__parameter__label',
+                children=p_label_name
+            ),
+            dcc.Input(
+                id='{}-input'.format(p_name),
+                className='app__parameter__input',
+                type='number',
+                value=p_value,
+                **kwargs
+            ),
+            html.Div(
+                id='{}-unit'.format(p_name),
+                className='app__parameter__unit',
+                children=p_unit
+            ),
+        ]
+    )
+
+
+def parse_upload_contents(contents, filename):
+    content_type, content_string = contents.split(',')
+    decoded = base64.b64decode(content_string)
+    df = pd.DataFrame()
+    try:
+        if 'csv' in filename:
+            # Assume that the user uploaded a CSV file
+            df = pd.read_csv(io.StringIO(decoded.decode('utf-8')))
+        elif 'xls' in filename:
+            # Assume that the user uploaded an excel file
+            df = pd.read_excel(io.BytesIO(decoded))
+    except Exception as e:
+        print(e)
+
+    return df.to_json()
+
+
+app.layout = html.Div(
+    id='app-div',
+    className='app',
+    children=[
+        dcc.Store(
+            id='data-store',
+            storage_type='session',
+            data=VDI_DATA.copy()
+        ),
+        html.Div(
+            id='top-panel-div',
+            className='app__container',
+            children=[
+                html.Div(
+                    id='timeseries-div',
+                    className='app__timeseries',
```

Comment:
these `className` properties are directely related to the definitions in the assets/base.css file.